ortho_seq_code/utils.py

```diff
@@ -11,9 +12,112 @@ def get_seq_info(seqf):
     sites = max(seq_series.str.len())
     pop_size = len(seq_series)
     seq_list = list(np.unique(list("".join(list(seq_series)))))
-    dm = len(seq_list)
     for i in seq_series:
         if i == "\n":
             pop_size -= 1
             seq.remove(i)
-    return [dm, sites, pop_size, seq, seq_series]
+    # Autopadding lowercase n's
+    if len(min(seq, key=len)) != len(max(seq, key=len)):
+        incomplete_seq_series = seq_series[seq_series.str.len() < sites]
+        while len(incomplete_seq_series) > 0:
+            incomplete_seq_series = seq_series[seq_series.str.len() < sites]
+            seq_series[incomplete_seq_series.index] += "n"
+        seq_series += "\n"
+        seq = list(seq_series)
+
+    # Custom alphabet
+    seq_oneline = "".join(seq_series)
+    seq_list = list("".join(list(seq_series)))
+    if alphbt_input is not None:
+        alphbt = alphbt_input.upper()
+        if alphbt == "PROTEIN_PNP":
+            alphbt_input = "RNDCEQHKSTY,"
+        elif alphbt == "ESSENTIAL":
+            alphbt_input = "ILVFWHKTM,"
+        elif alphbt == "ACIDIC":
+            alphbt_input = "DE,RHK,"
+        elif alphbt == "HYDROPHOBIC":
+            alphbt_input = "AVLIPFC,"
+        elif alphbt == "FREQUENCY":
```

Comment:
We can rename this to **FREQUENCY_11AA** as we'll also have the other frequency-based category (9 amino acid one).